File: article-service-ui/src/main/java/com/sciam/kogito/dto/Payment.java
package com.sciam.kogito.dto;


import io.quarkus.hibernate.orm.panache.PanacheEntity;
import jakarta.persistence.Entity;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.UUID;


@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Payment {
    private UUID transactionId;
    private double amount;
    private LocalDateTime paymentDate;
    private long orderId;
    private String countryOrigin;
    private String countryDestination;
    private PaymentMethod method;
    private PaymentStatus paymentStatus;
    private String reason;
    private String processInstanceId;
}

File: article-service-ui/src/main/webui/src/app/_interface/payment.ts
import {PaymentMethod} from "./enum/payment.method";

export interface Payment {
  transactionId: String;
  amount: number;
  paymentDate: Date;
  orderId: number;
  countryOrigin: string;
  countryDestination: string;
  method: PaymentMethod;
  reason: string;
}
